### PR TITLE
Fixes `get` and `set` not parsing in object literal shorthand syntax

### LINKED ISSUE #320: `{get}` should be a valid object literal
We currently throw a syntax error for the following case:

```js
let get = 1;
let obj = { get };
```

### PR DESCRIPTION
**Pull request by ianwjhalliday**
Fixes #320

Also renamed wellKnownPropertyPids.getter/setter to get/set to reflect
the actual PID they represent.

### CODE REVIEW

**Comment by ianwjhalliday** [comment]
@kunalspathak @boingoing @akroshg @aneeshdk please review
cc @abchatra @bterlson

**Comment by kunalspathak** [comment]
:+1:

**Comment by aneeshdk** [comment]
LGTM

**Comment by kunalspathak** [comment]
@ianwjhalliday , once merged to master, please port it to release/1.1 branch and bump up the QFE version number.

**Comment by ianwjhalliday** [comment]
Updated with fixes and refactored the get and set cases together into one if block.

File: test/es6/objlit.js [hunk 1]
```diff
@@ -15,15 +15,58 @@ var tests = [
...
+            var a = 0;
+            var get = 1;
+            var set = 2;
+            var z = 3;
+
+            var o = { get };
+            var p = { set };
+            var q = { get, set };
+            var r = { set, get };
+            var s = { get, z };
```
Comment (akroshg) [inline test/es6/objlit.js]: How about {get:get}?

Comment (ianwjhalliday) [inline test/es6/objlit.js]: I did a grep of the tests and found that `{ get: ... }` is a very common pattern and therefore well tested already.  `get` appearing after the colon is the same as it appearing in any expression context and I figure is also already well tested.

Comment (akroshg) [inline test/es6/objlit.js]: sounds good. This `var {get} = {}`  will not be error after your change, right?

Comment (ianwjhalliday) [inline test/es6/objlit.js]: Correct.  I added a new test for `get` and `set` name use in object destructuring to `test/es6/destructuring_obj.js`

Comment (akroshg) [inline test/es6/objlit.js]: Looks good.

**Comment by ianwjhalliday** [comment]
Rebased the fix bugs commit as a fixup

**Comment by ianwjhalliday** [comment]
@pleath would you mind taking a look at this one?

**Pull request merged**